src/CollisionEventsSubsystem.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+
+#include "CollisionEventsSubsystem.h"
+
+std::vector<std::pair<int, int>> CollisionEventsSubsystem::consume_collisions() {
+    auto ret = collisions_;
+    collisions_.clear();
+    return ret;
+}
+
+void CollisionEventsSubsystem::onConstraintBreak(
+    physx::PxConstraintInfo* constraints,
+    physx::PxU32 count
+) {
+    // do nothing
+}
+
+void CollisionEventsSubsystem::onWake(
+    physx::PxActor** actors,
+    physx::PxU32 count
+) {
+    // do nothing
+}
+
+void CollisionEventsSubsystem::onSleep(
+    physx::PxActor** actors,
+    physx::PxU32 count
+) {
+    // do nothing
+}
+
+void CollisionEventsSubsystem::onContact(
+    const physx::PxContactPairHeader& pairHeader,
+    const physx::PxContactPair* pairs,
+    physx::PxU32 nbPairs
+) {
+    collisions_.emplace_back(0, 0);
```

Comment:
I'm guessing the two 0's here should be representing the object ids involved in the collision? If so, can you please just comment it for now.

Reply:
Yea, it should be. But I had a really hard time getting the object ids from the given variables in this callback.